core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -183,7 +187,7 @@ private boolean hasPlayer() {
     }
 
     void start() {
-        if (isInPlaybackState()) {
+        if (playbackStateChecker.isInPlaybackState(mediaPlayer, currentState)) {
```

Comment:
Delegate checking of the playback state to a collaborator so that we can simply return `true` or `false` in our tests without needing to setup the media player to be in a certain state.